compiler/cli/cli-common/src/org/jetbrains/kotlin/utils/parametersMap.kt

```diff
@@ -218,10 +219,31 @@ private class AnyArgsConverter : ArgsConverter<Any> {
                 else -> null
             }
 
-        if (value::class.isSubclassOf(parameter.type.jvmErasure)) return ArgsConverter.Result.Success(value)
+        fun evaluateValue(arg: Any): Any? {
+            if (arg::class.isSubclassOf(parameter.type.jvmErasure)) return arg
```

Comment:
Are you sure that this line cannot return a valid `null` value? I'm not 100% sure about it, and if can, the logic below is incorrect.

Reply:
This logic was already there before. The case with null is handled in line 206 which I haven't changed